packages/langium/test/grammar/type-system/inferred-types.test.ts

```diff
@@ -889,10 +892,12 @@ describe('types of `$container` property are correct', () => {
         expect(interfacesString).toBe(s`
             export interface A extends AstNode {
                 readonly $container: D | E;
+                readonly $type: 'C';
```

Comment:
This doesn't seem to be correct. When parsing rule `A` in this grammar, we would expect an element with a `$type` with value `A` not `C`. The `$type` therefore needs to be `'A' | 'C'`.

Also, interface `D` should have `$type: 'D'` and `E` should have `$type: 'E'`.